include/engine/data_watchdog.hpp

```diff
@@ -38,102 +52,45 @@ class DataWatchdog
         return storage::SharedMemory::RegionExists(storage::CURRENT_REGION);
     }
 
-    using RegionsLock =
-        boost::interprocess::sharable_lock<boost::interprocess::named_sharable_mutex>;
-    using LockAndFacade = std::pair<RegionsLock, std::shared_ptr<datafacade::BaseDataFacade>>;
-
-    // This will either update the contens of facade or just leave it as is
-    // if the update was already done by another thread
-    LockAndFacade GetDataFacade()
+    auto GetDataFacade() const
     {
-        const boost::interprocess::sharable_lock<boost::interprocess::named_upgradable_mutex> lock(
-            shared_barriers->current_region_mutex);
+        return facade;
+    }
 
-        const auto shared_timestamp =
-            static_cast<const storage::SharedDataTimestamp *>(shared_regions->Ptr());
+  private:
 
-        const auto get_locked_facade = [this, shared_timestamp](
-            const std::shared_ptr<datafacade::SharedMemoryDataFacade> &facade) {
-            if (current_timestamp.region == storage::REGION_1)
-            {
-                return std::make_pair(RegionsLock(shared_barriers->region_1_mutex), facade);
-            }
-            else
-            {
-                BOOST_ASSERT(current_timestamp.region == storage::REGION_2);
-                return std::make_pair(RegionsLock(shared_barriers->region_2_mutex), facade);
-            }
-        };
+    void Run()
+    {
+        auto shared_memory = makeSharedMemory(storage::CURRENT_REGION);
```

Comment:
Why is the lock necessary here? All we do is to attach to a pre-defined shared memory location (that needs to exist prior to `libosrm` starting up) and then get a pointer `current` to the data contained in it. I don't see any actual reading before  `while (active && timestamp == current->timestamp)` right?